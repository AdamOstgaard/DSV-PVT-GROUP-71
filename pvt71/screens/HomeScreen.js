import React from "react";
import { StyleSheet, View, Alert, Text, Button } from "react-native";

import { Timer } from "../components/Timer";
import TimerSleepButton from "../components/TimerSleepButton";
import moment from "moment";
import { playSound } from "../SoundPlayer";

export default class HomeScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      timerPaused: false
    };
  }
  static navigationOptions = {
    header: null
  };

  render() {
    let pauseText;
    if (this.state.timerPaused) {
      pauseText = "VILOLÄGET ÄR PÅ";
    } else {
      pauseText = "VILOLÄGET ÄR AV";
    }

    return (
      <View style={styles.container}>
        <Text style={styles.sleepOnText}>{pauseText}</Text>
        <View style={styles.timerContainer}>
          <Timer
            onReset={reset => {
              this.setState({ timerPaused: reset });
            }}
            paused={this.state.timerPaused}
            style={styles.timer}
            startTime={moment.duration(2, "h").asMilliseconds()}
            callback={restart => {
              playSound();
              showRestartAlert(restart);
            }}
          />
        </View>
        <View style={styles.sleepButtonContainer}>
          <TimerSleepButton
            onResume={this.state.timerPaused}
            onToggle={enabled => this.setState({ timerPaused: enabled })}
          />
        </View>
        <Button title='Add contact' onPress={()=> {console.log('tr')}}></Button>
      </View>
    );
  }
}

const showRestartAlert = restart => {
  Alert.alert(
    "Timer",
    "Are you ok?",
    [
      { text: "Yes, I'm Ok", onPress: () => restart() },
      {
        text: "No, I need help!",
        onPress: () => console.log("Cancel Pressed"),
        style: "cancel"
      }
    ],
    { cancelable: false }
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "column",
    flex: 1,
    backgroundColor: "#fff"
  },
  timerContainer: {
    paddingTop: "10%",
    alignItems: "center"
  },
  sleepButtonContainer: {
    left: 240
  },
  sleepOnText: {
    marginTop: "30%",
    alignSelf: "center",
    fontSize: 18
  }
});
