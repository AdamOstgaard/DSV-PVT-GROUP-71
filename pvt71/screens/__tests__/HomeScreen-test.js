import "react-native";
import React from "react";
import HomeScreen  from "../HomeScreen";
import renderer from "react-test-renderer";

describe("Wizard Verify Contact Screen tests", () => {
    describe("Snapshot test", () => {
        it("renders correctly", () => {
        const tree = renderer.create(<HomeScreen />).toJSON();
        expect(tree).toMatchSnapshot();
        });
    });
});