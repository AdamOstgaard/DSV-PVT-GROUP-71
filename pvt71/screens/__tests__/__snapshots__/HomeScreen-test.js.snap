// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen test Snapshot test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
      "marginTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "textAlign": "left",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#0d6f73",
          "height": 60,
          "opacity": 1,
          "paddingLeft": 10,
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 30,
            "paddingRight": 10,
            "paddingTop": 8,
            "textAlign": "left",
          }
        }
      >
        Hjälp
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingTop": 1,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#0d6f73",
            "height": 60,
            "opacity": 1,
            "paddingLeft": 10,
            "textAlign": "left",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 30,
              "paddingRight": 10,
              "paddingTop": 8,
              "textAlign": "left",
            }
          }
        >
          Inställningar
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 6,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 18,
        }
      }
    >
      VILOLÄGET ÄR AV
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 1000,
          "borderStyle": "solid",
          "borderWidth": 2,
          "flexDirection": "column",
          "height": 280,
          "justifyContent": "center",
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": 280,
        }
      }
    >
      <Text
        callback={[Function]}
        onPress={[Function]}
        onReset={[Function]}
        paused={false}
        startTime={3600000}
        style={
          Object {
            "display": "flex",
            "fontSize": 64,
          }
        }
        warningCallback={[Function]}
        warningTime={1800000}
      >
        1:00:00
      </Text>
      <Text
        style={
          Object {
            "borderWidth": 0,
            "display": "flex",
            "fontSize": 12,
          }
        }
      >
        Tryck för att starta om
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 90,
            "opacity": 1,
            "width": 90,
          }
        }
      >
        <Image
          source={1}
        />
      </View>
    </View>
  </View>
</View>
`;
